AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,448 @@
+    param sc_graph_node_inspector: the sc graph node inspector qt widget
+    param sc_graph: the sc graph qt widget
+
+    returns: a list of node titles (i.e Print - Utilities/Debug). If there are duplicates of a node then the title
+    will include ( X Selected) in the string.
+    """
+    node_inspector_scroll_area = sc_graph_node_inspector.findChild(QtWidgets.QScrollArea, "")
+    # perform ctrl+a keystroke to highlight all nodes on the graph
+    QtTest.QTest.keyClick(sc_graph, "a", Qt.ControlModifier, WAIT_FRAMES_200)
+    node_inspector_backgrounds = node_inspector_scroll_area.findChildren(QtWidgets.QFrame, "Background")
+    titles = []
+    for background in node_inspector_backgrounds:
+        background_title = background.findChild(QtWidgets.QLabel, NODE_INSPECTOR_TITLE_KEY)
+        if background_title.text() != "":
+            titles.append(background_title.text())
+    return titles
+
+
+def get_sc_editor_node_inspector(sc_editor):
+    """
+    function for toggling the node inspector if it's not already turned on and returning the qt widget object
+
+    param sc_editor: the script canvas editor qt object
+
+    returns: the node inspector qt widget object
+
+    """
+    node_inspector_widget = sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT)
+    if sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT) is None:
+        action = pyside_utils.find_child_by_pattern(sc_editor, {"text": NODE_INSPECTOR_UI, "type": QtWidgets.QAction})
+        action.trigger()
+
+    return node_inspector_widget
+
+
+def create_script_event(self):
+    """
+    Function for creating a script event from the editor's asset editor.
+
+    param self: the script calling this function
+
+    returns None
+    """
+    action = pyside_utils.find_child_by_pattern(self.asset_editor_menu_bar, {"type": QtWidgets.QAction, "text": SCRIPT_EVENT_UI})
+    action.trigger()
+    result = helper.wait_for_condition(
+        lambda: self.asset_editor_row_container.findChild(QtWidgets.QFrame, EVENTS_QT) is not None, WAIT_TIME_3
+    )
+    Report.result(Tests.new_event_created, result)
+
+    # Add new child event
+    add_event = self.asset_editor_row_container.findChild(QtWidgets.QFrame, EVENTS_QT).findChild(QtWidgets.QToolButton, "")
+    add_event.click()
+    result = helper.wait_for_condition(
+        lambda: self.asset_editor_widget.findChild(QtWidgets.QFrame, DEFAULT_SCRIPT_EVENT) is not None, WAIT_TIME_3
+    )
+    Report.result(Tests.child_event_created, result)
+
+def create_script_event_parameter(self):
+    add_param = self.asset_editor_row_container.findChild(QtWidgets.QFrame, "Parameters").findChild(QtWidgets.QToolButton, "")
+    add_param.click()
+    result = helper.wait_for_condition(
+        lambda: self.asset_editor_widget.findChild(QtWidgets.QFrame, "[0]") is not None, WAIT_TIME_3
+    )
+    Report.result(Tests.parameter_created, result)
+
+def remove_script_event_parameter(self):
```

Comment:
Refactor to an asset editor uitls file or QtPy class.

Reply:
left a note on this to move it to the class once we implement it